doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,272 @@
+  allocate large block of time review a large PR.
+
+- Less wasted work if reviewers or maintainers reject the direction of the
+  change.
+
+- Easier to rebase and merge. Smaller PRs are less likely to conflict with other
+  changes in the tree.
+
+- Easier to revert if the PR breaks functionality.
+
+
+Defining Small PRs
+==================
+
+- Small PRs should encompass one self-contained logical change.
+
+- When adding a new large feature or API, the PR should address only one part of
+  the feature. In this case create an :ref:`RFC proposal <rfcs>` [#rfc]_ to
+  describe the additional parts of the feature for reviewers.
+
+- PRs should include tests or samples for any added or changed lines of code.
+
+- PRs must update any documentation affected by the functional code changes.
+
+- If introducing a new API, the PR must include an example usage of the API.
+  This provides context to the reviewer and prevents submitting PRs with unused
+  APIs.
+
+
+Multiple Commits on a Single PR
+===============================
+
+You are further encouraged to break up PRs into multiple commits.  Keep in mind
+each commit of your PR must still build cleanly and pass all the CI tests.
+
+For example, when introducing an extension to an API, you can break up the PR
+into multiple commits targeting these specific changes:
+
+#. Introduce the new APIs, including shared devicetree bindings
+#. Update driver implementation X, with driver specific devicetree bindings
+#. Update driver implementation Y
+#. Add tests for the new API
+#. Add a sample using the API
+#. Update the documentation
+
+Large Changes
+=============
+
+Large changes to the Zephyr project must submit an :ref:`RFC proposal <rfcs>`
+describing the full scope of change and future work.  The RFC proposal provides
+the required context to reviewers, but allows for smaller, incremental, PRs to
+get reviewed and merged into the project. The RFC should also define the minimum
+viable implementation.
+
+Changes which require an RFC proposal include:
+
+- Submitting a new feature.
+- Submitting a new API.
+- :ref:`treewide-changes`.
+- Other large changes that can benefit from the RFC proposal process.
+  Maintainers have the discretion to request you create an RFC for your change.
+
+PR Requirements
+***************
+
+- Each commit in the PR must provide a commit message following the
+  :ref:`commit_message_guidelines`.
+
+- All files in the PR must comply with :ref:`Licensing
+  Requirements<licensing_requirements>`.
+
+- Follow the Zephyr :ref:`coding_style` and :ref:`coding_guidelines`.
+
+- PRs must pass all CI checks. This is a requirement to merge the PR. You may
+  mark a PR as draft and explicitly request reviewers to provide early feedback,
+  even with failing CI checks.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. The
+  CI system does not enforce this policy, so it is the PR author's
+  responsibility to verify.
```

Comment:
Would not go as far as suggesting that considering how long each run takes. I understand it's just a suggestion but people unfamiliar with the process would end up doing it all the time and burning half a day on it, it sounds intimidating for no good reason (building for one specific case should be good enough).

Reply:
> Would not go as far as suggesting that considering how long each run takes. [...]  and burning half a day on it, 

This is of course done in _parallel_. 

Clarifying:

"In doubt, this verification can be performed by submitting OTHER, even smaller, temporary, test draft PR(s) without any reviewer and closing them unmerged after they passed the CI tests."

> people unfamiliar with the process would end up doing it all the time [...] it sounds intimidating for no good reason 

"In doubt".

In my fairly extensive experience asking people to submit test PRs, including people unfamiliar with github
- they never look spend time having  extra "fun" with github unless they're asked  or unless it makes sense
- when asked, they all immediately  understand that it can and should be done in a different PR.


More generally speaking, _the practice of tests draft PRs should really be encouraged and developed_. There is a different use case where test draft PRs "save lives" and regularly find (or: could have found!) regressions before merge: when running the CI of one git repo on a code change in a git different repo: 

- https://docs.zephyrproject.org/latest/develop/modules.html#process-for-submitting-changes-to-existing-modules
- https://github.com/zephyrproject-rtos/zephyr/pull/53327#issuecomment-1414193515
- https://github.com/thesofproject/rimage/commit/6e22944e05ea7c6d6c2b61e9632bdbf017644323